api/queries_pr.go

```diff
@@ -672,6 +672,7 @@ func PullRequestForBranch(client *Client, repo ghrepo.Interface, baseBranch, hea
 					number
 					title
 					state
+					closed
```

Comment:
The fix looks good, but since the concept of "closed" is ambiguous for pull requests, and we're already fetching the `state` field, I would say let's go a few extra steps and rely just on the `state` field and avoid fetching the `closed` property altogether.

The two non-open states of Pull Requests are "closed" and "merged". Unless explicitly clarified, it can be ambiguous whether "closed" includes "merged" or not. For instance, `pr.State == "CLOSED"` does not include the "merged" state.

To avoid this ambiguity in our Go code, let's quit querying the `closed` field and change all `pr.Closed` checks (3 that I count right now) to actually reflect what they are checking for: the not-open state `!pr.IsOpen()`. That would involve adding an extra `IsOpen()` method to PullRequest, but I think it will be worth it. The benefit is that we can fetch one fewer field in GraphQL. What do you think?

Reply:
Definitely agree that it's a clearer and more efficient approach, thanks!

Just for reference, per [the GraphQL API docs](https://docs.github.com/en/graphql/reference/enums#pullrequeststate), `CLOSED` means that a PR was closed without being merged.